google/gax/api_callable.py

```diff
@@ -382,6 +382,28 @@ def inner(*args, **kwargs):
     return inner
 
 
+def _merge_options_metadata(options, settings):
```

Comment:
> That logic merges kwargs dictionary, just overwriting values, which share the same key.

Sure, but if we always want "deep merging" for metadata when we merge options and settings anyway, why not just do the deep merging in the existing method? Behaviorally, that's what your code already does.

> the merging is not necessarily about x-goog-api-client header, but also about some other headers (there will be more, as part of the "header regionalization" project i'm working on now).

I think the key part of the `construct_settings` merge is that `x-goog-api-client` is special: it should come last. (@lukesneeringer Correct me if I'm wrong.) It seems like this method violates that.

I am asking if it would make more sense to create a metadata merge method that is reused in both places and special-cases `x-goog-api-client` correctly.